ChemistryLib/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+#Source files grouped by a directory
+get_source_files(SOURCES)
+append_source_files(SOURCES PhreeqcInterfaceData)
+
+# Create the library
+add_library(ChemistryLib ${SOURCES})
+if(BUILD_SHARED_LIBS)
+    install(TARGETS ChemistryLib LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+endif()
+
+include(GenerateExportHeader)
+generate_export_header(ChemistryLib)
+
+target_link_libraries(ChemistryLib
+    PUBLIC iphreeqc
+    PRIVATE NumLib # ParameterLib
```

Comment:
@bilke The `ProcessLib::ProcessVariable::getName()` is defined in the ProcessLib, which is not listed as dependency of the ChemistryLib. That's the reason for the failing linking.

Reply:
@endJunction Yes, but when linking with ProcessLib we have a cyclic dependency.

Reply:
The problem is in the design of the ChemistryLib, which must not require additional knowledge about processes or process variables---think of it as a set of complicate constitutive equations.
Essentially the IPhreeqc interface should be an independent process (*i.e.* be derived from Process class) and also not intrude the global time-loop.

Reply:
Cyclic dependency issue seems resolved.

Reply:
- Why the global time-loop is intrude.
@endJunction @HBShaoUFZ 
It is improper to place `doWaterChemistryCalculation` inside `Process::postIteration` and `Process::postNonLinearSolver`, as these functions are called in the loop over individual processes when solving a particular coupled process. After all the coupled processes are solved, we need to call Phreeqc to do speciation calculation by means of `doWaterChemistryCalculation`. This part of work can be thought to be done in the post-time-step stage. In this stage, `Process::postTimestep` is invoked as the function named, while it is inside the loop over individual processes. Thus, `doWaterChemistryCalculation` is placed here before the loop for data processing in the post-time-step stage.